### PR TITLE
Add log functionality

### PR DESCRIPTION
**Pull request by apanakkat**

### CODE REVIEW

File: fore/foresight/client.py [hunk 1]
```diff
@@ -178,6 +180,47 @@ def generate_answers_and_run_eval(self, generate_fn: GenerateFnT,
...
+        Args:
+            query: The query for evaluation.
+            llm_response: The response generated by the llm against the query.
+            contexts: List of contexts relevant to the query.
+        """
+        inference_output = InferenceOutput(
+            generated_response=llm_response, contexts=contexts)
+        log_entry = LogTuple(query=query, inference_output=inference_output)
+        self.log_entries.append(log_entry)
+
```
Comment (forefab) [inline fore/foresight/client.py:197]: should we add an autoflush configuration? e.g. if len(log_entries) > self.auto_flush_size

File: fore/foresight/client.py [hunk 2]
```diff
@@ -178,6 +180,47 @@ def generate_answers_and_run_eval(self, generate_fn: GenerateFnT,
...
+            method="put",
+            endpoint="/api/eval/log",
+            input_json=log_request.model_dump(mode="json"))
+
+        if response.status_code == 200:
+            logging.info(
+                "Log entries flushed successfully. Visit %s to view results.",
+                self.ui_url)
+
+        # Clear log entries after flushing
```
Comment (forefab) [inline fore/foresight/client.py:219]: If response_code != 200 we shouldn't clear the log_entries - maybe they will flush next time.

File: fore/foresight/client.py [hunk 3]
```diff
@@ -178,6 +180,47 @@ def generate_answers_and_run_eval(self, generate_fn: GenerateFnT,
 
         return response
 
+    def log(self, query: str, llm_response: str, contexts: list[str]) -> None:
```
Comment (forefab) [inline fore/foresight/client.py:183]: s/llm_response/response/ ?

it could be a RAG response, or a decision tree response ;)

**Review by forefab — APPROVED** [review]

File: fore/foresight/client.py (hunk 1 above)
Comment (apanakkat) [reply fore/foresight/client.py:197]: Great idea. Done.

File: fore/foresight/client.py (hunk 2 above)
Comment (apanakkat) [reply fore/foresight/client.py:219]: Good point.

**Comment by apanakkat** [comment]
Thanks for the review, merging!

**Pull request merged**